covid-vaccine-faq-bot/functions/es-dialogflow-detect-intent.js

```diff
@@ -0,0 +1,45 @@
+// Imports the Google Cloud API library
+const {SessionsClient} = require('@google-cloud/dialogflow');
+ 
+exports.handler = async function (context, event, callback) {
+ 
+ let query = event.utterance;
+ 
+ // google requires an environment variable called GOOGLE_APPLICATION_CREDENTIALS that points to a file path with the service account key file (json) to authenticate into their API
+ // to solve for this, we save the key file as a private asset, then use a helper function to find and return the path of the private asset.
+ // lastly we set the environment variable dynamically at runtime so that it's in place when the sessions client is initialized
+//process.env.GOOGLE_APPLICATION_CREDENTIALS = Runtime.getAssets()['/service-account-key.json'].path;
```

Comment:
Shouldn't we use the environment variable here in case someone changes the path where the key is located?

`Runtime.getAssets()[context.GOOGLE_APPLICATION_CREDENTIALS].path;`

Reply:
Updated with a new mechanism for using GCP creds.